llvm/test/CodeGen/PowerPC/aix-small-tls-globalvarattr-funcattr.ll

```diff
@@ -0,0 +1,197 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 3
+; RUN: llc  -verify-machineinstrs -mcpu=pwr7 -ppc-asm-full-reg-names \
+; RUN:      -mtriple powerpc64-ibm-aix-xcoff < %s \
+; RUN:      | FileCheck %s --check-prefix=CHECK-SMALLCM64
+; RUN: llc  -verify-machineinstrs -mcpu=pwr7 -ppc-asm-full-reg-names \
+; RUN:      -mtriple powerpc64-ibm-aix-xcoff --code-model=large \
+; RUN:      < %s | FileCheck %s --check-prefix=CHECK-LARGECM64
+
+@mySmallLocalExecTLS6 = external thread_local(localexec) global [60 x i64], align 8
+@mySmallLocalExecTLS2 = external thread_local(localexec) global [3000 x i64], align 8 #0
+@MyTLSGDVar = thread_local global [800 x i64] zeroinitializer, align 8
+@mySmallLocalExecTLS3 = internal thread_local(localexec) global [3000 x i64] zeroinitializer, align 8
+@mySmallLocalExecTLS4 = internal thread_local(localexec) global [3000 x i64] zeroinitializer, align 8 #0
+@mySmallLocalExecTLS5 = thread_local(localexec) global [3000 x i64] zeroinitializer, align 8 #0
+@mySmallLocalExecTLS = thread_local(localexec) local_unnamed_addr global [7800 x i64] zeroinitializer, align 8 #0
```

Comment:
general question for all the test cases in the patch. do we still need so many thread_local(localexec) variable to test  the feature, can we reduce the test case?  for example : 
`@mySmallLocalExecTLS = thread_local(localexec) local_unnamed_addr global [7800 x i64] zeroinitializer, align 8 #0`

but never check it in the test case ?

Reply:
It's not a necessity to have this many variables, but I just wanted a mixture of large variables, with either external/internal/regular thread_local to test the feature. You're right though, I missed that `@mySmallLocalExecTLS` isn't being used here so I probably  don't need it and can remove it.

Reply:
most of test scenario is duplicated with the PR  https://github.com/llvm/llvm-project/pull/71485 , I have different option of  duplicated test, It may be increase the cost maintain of the test case , if the functionality is changed some day, we have to change all the the test case which test the some functionality.

Reply:
Fair point from the maintenance perspective. I will try to reduce this a bit more.